src/Files.App/Extensions/MessageFormatExtensions.cs

```diff
@@ -15,7 +15,7 @@ public static class MessageFormatExtensions
 		private static readonly ResourceMap resourcesTree = new ResourceManager().MainResourceMap.TryGetSubtree("Resources");
 
 		// CultureInfo based on the application's primary language override
-		private static readonly CultureInfo locale = new(ApplicationLanguages.PrimaryLanguageOverride);
+		private static readonly CultureInfo locale = ApplicationLanguages.PrimaryLanguageOverride == string.Empty ? CultureInfo.InstalledUICulture : new(ApplicationLanguages.PrimaryLanguageOverride);
```

Comment:
If `ApplicationLanguages.PrimaryLanguageOverride` is empty, we can check if the system language is in the list of supported languages. Otherwise, we can fall back to en--us.

I don't think we need to modify `GeneralViewModel.cs`.

Reply:
Agree

Reply:
@XTorLukas what are your thoughts? In theory, this should be the cleanest solution.

Reply:
It looks good, it's just very primitive, but a future helper will add better control over application languages and also the ability to see if I'm compatible with the system language, otherwise use en-US.

Reply:
The reason for the change to `GeneralViewModel.cs` was to store the language name, thus avoiding string.Empty, but this is also addressed in a future helper class